### DISCUSSION TITLE
Execute postUpgradeTasks in directory of package file?

### DISCUSSION THREAD

**Discussion by fgreinacher** [renovatebot/renovate#8274] [Request Help] [answered]
We have a directory structure similar to this:

```
repo
|-- package-a
      |-- package.json
|-- package-b
      |-- package.json
|-- package-c
      |-- package.json
```

And want to run a custom tool in the `package-*` directories after the relevant package files have been changed.

Is this currently possible with the `postUpgradeTasks` mechanism?

**Comment by fgreinacher** [comment]
Do you have any thoughts on this @viceice, @rarkins or @zharinov?

**Reply by rarkins** [reply]
I think that `baseDir` is an exposed config option. Could you allow templating and configure `cd {{baseDir}} && <command>` ?

**Reply by fgreinacher** [reply]
Yes, that's working perfectly, thanks!

**Comment by fgreinacher** [comment, marked as answer]
Quoting @rarkins response that worked great for me:

> I think that baseDir is an exposed config option. Could you allow templating and configure cd {{baseDir}} && <command> ?

**Reply by rarkins** [reply]
FYI I am renaming `baseDir` to `packageFileDir` due to a naming conflict but it should get migrated automatically: https://github.com/renovatebot/renovate/pull/8625
